### PR TITLE
EVG-5684 refactor inputs to host allocation to be public

### PR DESCRIPTION
**Pull request by StuartButler**

### CODE REVIEW

File: scheduler/utilization_based_host_allocator.go [hunk 1]
```diff
@@ -221,27 +159,66 @@ func evalHostUtilization(ctx context.Context, d distro.Distro, taskQueue []model
...
+			data.Hosts = append(data.Hosts, taskGroupData.Hosts...)
+			taskGroupDatas[name] = data
+		}
 	}
-	return scheduledTasksDuration
+
+	return taskGroupDatas
+}
+
+func makeTaskGroupString(group, bv, project, version string) string {
```
Comment (tychoish) [inline scheduler/utilization_based_host_allocator.go]: feel like this function should probably be a method on the model?

File: scheduler/duration_based_host_allocator.go [hunk 2]
```diff
@@ -1,495 +1,490 @@
...
-		"current_tasks_runtime":        curTasksRuntime,
-		"current_tasks_runtime_span":   curTasksRuntime.String(),
-		"queue_length":                 len(taskQueueItems),
-		"scheduled_tasks_runtime":      schTasksRuntime,
-		"scheduled_tasks_runtime_span": schTasksRuntime.String(),
-	})
-
-	return numNewHosts, nil
-}
+// const (
```
[the remark this answers is not in the document]
Comment (StuartButler) [reply scheduler/duration_based_host_allocator.go]: I just wanted confirmation that there was definitely no further need for this allocator and its associated tests.  I'll go ahead and delete *duration_based_host_allocator.go* and *duration_based_host_allocator_test.go*

File: scheduler/wrapper.go [hunk 3]
```diff
@@ -17,6 +18,83 @@ import (
...
+const (
+	// maximum turnaround we want to maintain for all hosts for a given distro
+	MaxDurationPerDistroHost               = 30 * time.Minute
+	MaxDurationPerDistroHostWithContainers = 2 * time.Minute
+	dynamicDistroRuntimeAlertThreshold     = 24 * time.Hour
+)
+
+// Should TaskGroupInfo and DistroQueueInfo live in the model package?
+type TaskGroupInfo struct {
+	Name                  string        `bson:"name" json:"name"`
```
[the remark this answers is not in the document]
Comment (StuartButler) [reply scheduler/wrapper.go]: Moving *TaskGroupInfo* and *DistroQueueInfo* to the *model* package.

File: scheduler/wrapper.go [hunk 4]
```diff
@@ -17,6 +18,83 @@ import (
...
+
+type DistroQueueInfo struct {
+	Length             int             `bson:"length" json:"length"`
+	ExpectedDuration   time.Duration   `bson:"expected_duration" json:"expected_duration"`
+	CountOverThreshold int             `bson:"count_over_threshold" json:"count_over_threshold"`
+	TaskGroupInfos     []TaskGroupInfo `bson:"task_group_infos" json:"task_group_infos"`
+	taskGroupInfosMap  map[string]TaskGroupInfo
+}
+
+func GetDistroQueueInfo(tasks []task.Task, maxDurationThreshold time.Duration, distro distro.Distro) DistroQueueInfo {
```
[the remark this answers is not in the document]
Comment (StuartButler) [reply scheduler/wrapper.go]: Oops.  The *distro.Distro* is actually not used within the function -- removing.

File: scheduler/duration_based_host_allocator_test.go [hunk 5]
```diff
@@ -1,1226 +1,1292 @@
...
-			// 'one-host-per-scheduled-task': 3
-			newHosts, err := durationNumNewHostsForDistro(ctx, hostAllocatorData,
-				tasksAccountedFor, distroScheduleData)
-			So(err, ShouldBeNil)
-			So(newHosts, ShouldEqual, 2)
-		})
-	})
-
-}
+// func TestDurationBasedNewHostsNeeded(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (StuartButler) [reply scheduler/duration_based_host_allocator_test.go]: Deleted.

File: scheduler/utilization_based_host_allocator.go (hunk 1 above)
Comment (StuartButler) [reply scheduler/utilization_based_host_allocator.go]: Added separate functions to *task.Task* and *host.Host*

File: model/distroqueue/distro_queue.go [hunk 6]
```diff
@@ -0,0 +1,21 @@
+package distroqueue
```
Comment (tychoish) [inline model/distroqueue/distro_queue.go]: is this a new package? 

maybe it makes sense to put this in top-level model directory with the other task queue structure

File: scheduler/duration_based_host_allocator.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (john-m-liu) [reply scheduler/duration_based_host_allocator.go]: Yep. Also if we needed it again for some reason, the history is stored in github

File: model/distroqueue/distro_queue.go (hunk 6 above)
Comment (StuartButler) [reply model/distroqueue/distro_queue.go]: Moved to the *model* package.

**Pull request merged**